src/mem/mem_ctrl.cc

```diff
@@ -335,18 +573,31 @@ MemCtrl::addToWriteQueue(PacketPtr pkt, unsigned int pkt_count, bool is_dram)
         // and enqueue it
         if (!merged) {
             MemPacket* mem_pkt;
+
+            // COMMENT: Here we need to determine how to handle this for
+            // DRAM cache
+            // the mem_pkt returned has interface specific things (different)
+            // for different devices
+            // If this is not serviced by DRAM let's say, then you will need
+            // a way to re-write the packet in the write queue.
+
+            // COMMENT: Check if this packet is in DRAM cache through tags
             if (is_dram) {
                 mem_pkt = dram->decodePacket(pkt, addr, size, false, true);
+                mem_pkt->read_before_write = true;
                 dram->setupRank(mem_pkt->rank, false);
```

Comment:
Should this be true?

Reply:
I think this should be true for all writes